### PR TITLE
Doc: Add Mumur term to the glossary

### PR DESCRIPTION
**Pull request by tzach**

### CODE REVIEW

File: docs/glossary.rst [hunk 1]
```diff
@@ -180,3 +180,7 @@ Glossary
     Workload
       A database category that allows you to manage different sources of database activities, such as requests or administrative activities. By defining workloads, you can specify how ScyllaDB will process those activities. For example, you can prioritize one workload over another (e.g., user requests over administrative activities). See :doc:`Workload Prioritization </using-scylla/workload-prioritization>`.
 
+    MurmurHash3
+       A hash function `created by Austin Appleby <https://en.wikipedia.org/wiki/MurmurHash>`_, and used by the :term:`Partitioner` to distribute the partitions between nodes. 
+       The MurmurHash3 version used in ScyllaDB is originated from `Apache Cassandra <https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/digest/MurmurHash3.html>`_.
```
Comment (nyh) [inline docs/glossary.rst:185]: Maybe you want to common that Cassandra's variant is NOT identical to the official MurmurHash3 calculation. It is admitted in
https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/utils/MurmurHash.java#L31-L33
that this is a bug, that can't be fixed. The sign problem is explained in https://github.com/russss/murmur3-cassandra.

**Review by nyh — CHANGES_REQUESTED** [review]
Looks good but I would add a comment that our algorithm - because of Cassandra compatibility - is not actually the original MurmurHash3 algorithm but a slightly and unintentionally changed version of it (more details below).

File: docs/glossary.rst (hunk 1 above)
Comment (tzach) [reply docs/glossary.rst]: Good point. See latest version

File: docs/glossary.rst [hunk 2]
```diff
@@ -180,3 +180,8 @@ Glossary
     Workload
       A database category that allows you to manage different sources of database activities, such as requests or administrative activities. By defining workloads, you can specify how ScyllaDB will process those activities. For example, you can prioritize one workload over another (e.g., user requests over administrative activities). See :doc:`Workload Prioritization </using-scylla/workload-prioritization>`.
 
+    MurmurHash3
+       A hash function `created by Austin Appleby <https://en.wikipedia.org/wiki/MurmurHash>`_, and used by the :term:`Partitioner` to distribute the partitions between nodes.
+       The name comes from two basic operations, multiply (MU) and rotate (R), used in its inner loop.
+       The MurmurHash3 version used in ScyllaDB originated from `Apache Cassandra <https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/digest/MurmurHash3.html>`_, and is **not** identical to the `official MurmurHash3 calculation <https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/utils/MurmurHash.java#L31-L33>`_. More `here <https://github.com/russss/murmur3-cassandra.>`_.
```
Comment (annastuchlik) [inline docs/glossary.rst:186]:
```suggestion
       The MurmurHash3 version used in ScyllaDB originated from `Apache Cassandra <https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/digest/MurmurHash3.html>`_, and is **not** identical to the `official MurmurHash3 calculation <https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/utils/MurmurHash.java#L31-L33>`_. More `here <https://github.com/russss/murmur3-cassandra>`_.
```

**Comment by tzach** [comment]
@nyh I rebase the PR with all updates. Please review

**Review by annastuchlik — APPROVED** [review]
LGTM

**Comment by annastuchlik** [comment]
@nyh Could you re-review this PR?

**Comment by tzach** [comment]
@nyh Could you re-review this PR?

**Comment by annastuchlik** [comment]
@nyh ping

**Comment by tzach** [comment]
@nyh can you please review

**Review by nyh — APPROVED** [review]

**Pull request closed without merging**